crypto/keys/sr25519/sr25519.go

```diff
@@ -0,0 +1,151 @@
+package sr25519
```

Comment:
Love seeing this land. It's out of scope of this PR but we should open an issue outlining a batch verification API.

Reply:
Does it make sense to do batch verification for transactions? Compared to blocks, we don't have hundreds signers for the same transaction.

Reply:
Even with blocks we don't have signers with the same message.

Reply:
I think batch verification only works for same signature message, right?

Reply:
No, batch verification still gives notable speedups across distinct messages for both pairings, and non-pairing signature schemes